packages/ts/generator-plugin-signals/src/SignalProcessor.ts

```diff
@@ -47,293 +49,195 @@ export default class SignalProcessor {
+        return (
+          imports.named.getIdentifier(HILLA_LIT_FORM, 'ArrayModel') ?? imports.named.add(HILLA_LIT_FORM, 'ArrayModel')
+        );
       }
-    }
-    throw new Error('Unsupported type reference node');
-  }
 
-  #getExistingEntityModelUniqueIdentifier(modelName: string) {
-    const { imports } = this.#dependencyManager;
-    return (
-      imports.named.getIdentifier('@vaadin/hilla-lit-form', modelName) ??
-      imports.default.iter().find(([path]) => path.endsWith(`/${modelName}.js`))?.[1]
-    );
-  }
+      // Otherwise, we calculate and import the model class.
+      const [path] = Iterator.from(imports.default).find(([, id]) => id === node) ?? [];
 
-  #addModelImport(
-    entityName: string | undefined,
-    modelName: string | undefined,
-    modelNameUniqueId: ts.Identifier | undefined,
-  ) {
-    if (modelName) {
-      if (primitiveModels.values().find((primitiveModel) => primitiveModel === modelName)) {
-        const { imports } = this.#dependencyManager;
-        const importedModel = imports.named.getIdentifier('@vaadin/hilla-lit-form', modelName);
-        if (importedModel === undefined) {
-          imports.named.add('@vaadin/hilla-lit-form', modelName, false, modelNameUniqueId);
-        }
-      } else {
-        this.#addObjectModelImport(entityName!, modelName, modelNameUniqueId!);
+      if (!path) {
+        throw new Error(`Model not found for ${node.text}`);
       }
-    }
-  }
 
-  #addObjectModelImport(entityName: string, modelName: string, modelNameUniqueId: ts.Identifier) {
-    const { imports } = this.#dependencyManager;
-    const entityImport = imports.default
-      .iter()
-      .map(([path]) => path)
-      .find((path) => path.startsWith('./') && path.endsWith(`/${entityName}.js`));
-    if (entityImport) {
-      const entityModelImportPath = entityImport.replace(`/${entityName}.js`, `/${modelName}.js`);
-      const importedModel = imports.default.paths().find((path) => path === entityModelImportPath);
-      if (importedModel === undefined) {
-        imports.default.add(entityModelImportPath, modelName, false, modelNameUniqueId);
+      const modelName = `${node.text}Model`;
+      const modelPath = `${dirname(path)}/${modelName}${extname(path)}`;
+
+      return imports.default.getIdentifier(modelPath) ?? imports.default.add(modelPath, modelName);
+    } else if (isTypeNode(node)) {
+      // If the node is a primitive type, we will import the corresponding
+      // model class from `@vaadin/hilla-lit-form`.
+      let modelName: string | undefined;
+
+      if (node.kind === SyntaxKind.ArrayType) {
+        modelName = 'ArrayModel';
+      } else if (node.kind === SyntaxKind.StringKeyword) {
+        modelName = 'StringModel';
+      } else if (node.kind === SyntaxKind.NumberKeyword) {
+        modelName = 'NumberModel';
+      } else if (node.kind === SyntaxKind.BooleanKeyword) {
+        modelName = 'BooleanModel';
+      } else {
+        return undefined;
```

Comment:
This could've remained the same as before where it was handled using a static mapping between the `SyntaxKind` and primitive type's model names.

Reply:
Sounds like a nano optimization 😅 But, `switch` looks better at least!